### ISSUE TITLE
Add new options, and keep current URI

### ISSUE THREAD

**Issue by kvdmolen** [vuejs/vue-router#461]
To avoid this:

``` js
var q = this.$route.query // q = {size: 10, view: 'grid'}
q.q = this.search
this.$router.go({name: this.$route.name, params: this.$route.params, query: q})
```

when you need to maintain the current route, it would be great to have something like:

``` js
this.$router.go({query: {q: this.search}, keep: true})
```

or

``` js
this.$router.add({query: {q: this.search}})
```

which maintains `{size: 10, view: 'grid'}` and becomes `{size: 10, view: 'grid', q: 'searchstring'}`

**Comment by nblackburn** [comment]
Beat me to it, was going to suggest this but didn't have time.

**Comment by simplesmiler** [comment]
I've suggested something similar in https://github.com/vuejs/vue-router/issues/437, but I think explicit `keep` is a better way to handle it.

**Comment by fnlctrl** [comment]
@kvdmolen FYI it's been implemented in 2.0 :)

**Comment by kvdmolen** [comment]
Yes I just found out 👍
